proposals/lip-0052.md

```diff
@@ -2295,51 +2239,47 @@ Let `genesisBlockAssetBytes` be the `data` bytes included in the block assets fo
-  - The `userSubstore` array must be in lexicographical order of `address`. For a given `address`, the entries must be in lexicographic order of `nftID`.
-  - Across all elements of the `escrowSubstore` array, the pairs `escrowedChainID`, `nftID` must be unique: each pair appears at most once, although a given `escrowedChainID` can appear multiple times with different unique `nftID`. Furthermore, each `nftID` should also be unique.
-  - For a given element of either the `userSubstore` or `escrowSubstore` arrays, there should exist a unique element in the `NFTSubstore` array, having the same value of the `nftID`.
   - The `supportedNFTsSubstore` array, if non-empty, must adhere to one of the following two mutually exclusive conditions:
     - Have a single entry with `chainID` of length `0` and no other entries. This indicates that all NFTs are supported, i.e., `chainID == ALL_SUPPORTED_NFTS_KEY`. In this case, the `supportedCollectionIDArray` should be an empty array.
     - Across all elements, the `chainID` must be unique and have length `LENGTH_CHAIN_ID`. Furthermore, the `supportedNFTsSubstore` array must be in lexicographic order of `chainID`. Also, for each entry of this array, the `supportedCollectionIDArray` should be in lexicographic order.
   
-- For each entry `NFTEntry` in `genesisBlockAssetObject.NFTSubstore`, create an entry in the NFT substore with:
-
-  ```python
-  storeKey = NFTEntry.nftID
-  storeValue = encode(
-      schema = NFTStoreSchema,
-      object = {
-          "owner": NFTEntry.owner,
-          "attributesArray": NFTEntry.attributesArray
-      }
-  )
-  ```
+- For each entry `NFTEntry` in `genesisBlockAssetObject.NFTSubstore`:
+  - Create an entry in the NFT substore with:
+
+    ```python
+    storeKey = NFTEntry.nftID
+    storeValue = encode(
+        schema = NFTStoreSchema,
+        object = {
+            "owner": NFTEntry.owner,
+            "attributesArray": NFTEntry.attributesArray
+        }
+    )
+    ```
 
-  Furthermore, if `NFTEntry.owner` has length `LENGTH_ADDRESS` bytes, create an entry in the user substore with:
+  - If `NFTEntry.owner` has length `LENGTH_ADDRESS` bytes, create an entry in the user substore with:
 
-  ```python
-  storeKey = NFTEntry.owner + NFTEntry.nftID
-  storeValue = encode(
-      schema = userStoreSchema,
-      object = {
-          "lockingModule": NFT_NOT_LOCKED
-      }
-  )
-  ```
-
-- For each entry `escrowEntry` in `genesisBlockAssetObject.escrowSubstore`, create an entry in the escrow substore with:
+    ```python
+    storeKey = NFTEntry.owner + NFTEntry.nftID
+    storeValue = encode(
+        schema = userStoreSchema,
+        object = {
+            "lockingModule": NFT_NOT_LOCKED
+        }
+    )
+    ```
+    maintaining the lexicographical ordering of the user substore by `NFTEntry.owner` and, for each `NFTEntry.owner`, by `NFTEntry.nftID`.
```

Comment:
What does it mean that we maintain the lexicographical order of the substore? The substore is basically just an hashmap, I don't think there is any order in this sense